Gs 00009 (#52)
bitsandvolts/bitsandvolts.github.io: _posts/2023-09-05-gs_00009_vs-code.md

---
layout: single
title: Visual Studio Code
category: Getting Started
permalink: /categories/getting_started/visual_code
excerpt: "Visual Studio Code - More than just an Editor"
description:
  Visual code is a modern IDE that is free to use and offers a high degree of
  customization. We show how to install VS code on Manjaro.
author: Bjoern Hartmann
toc: true
toc_label: "Content"
toc_sticky: true
toc_icon: "align-justify"
sidebar:
  nav: "getting_started_0"
header:
  overlay_color: "#000"
  overlay_filter: "0.5"
  overlay_image: /assets/images/getting_started.jpeg
---

Welcome to our next post! :-) Today's topic is a very prominent code editor:

> **Visual Studio Code** (VS Code). According to _Wakatime_, VS Code was the most

Reviewer: ```suggestion
**Visual Studio Code** (VS Code). According to _Wakatime_, VS Code was by far the most
```

> used code editor in 2022. And that's for a reason! VS Code offers a rich feature

Reviewer: ```suggestion
used code editor in 2022 (see the chart below). And that's for a reason! VS Code offers a rich feature
```

set at a high degree of customization. By the way, this post is written using VS
Code ;-) First, we want to discuss different kinds of editors. Then, we show you
how to install VS code step-by-step. As a benefit, we provide you our custom
settings.

<figure class="img-center" style="width:80%">
  <img src="/assets/images/gs_00009_top_editors.png" title="Top Editors 2022" alt="Top Editors 2022">
  <figcaption class="text-center">
    <a href="https://wakatime.com/blog/57-wakatime-2022-programming-stats">https://wakatime.com/blog/57-wakatime-2022-programming-stats</a>
  </figcaption>
</figure>

## Editors & IDEs

Nowadays, simple editors that only offer text editing are super rare. Even tiny
editors have features like **syntax highlighting** and **autocompletion**.
Syntax highlighting highlights keywords of a programming language. Soon, you
will notice that it is much easier to read colorized than simple black-and-white

> code. Autocompletion is a handy feature. Imagine you define a variable and want

Reviewer: ```suggestion
code. Autocompletion is a handy feature: Imagine you define a variable and want
```

to use it later. Once you start typing the first letters of the variable, it is
automatically completed. That saves a lot of typing.

> In general, there are two types of editors. Terminal editors like Vim or Nano

Reviewer: ```suggestion
In general, there are two types of editors. _Terminal_ editors like Vim or Nano
```

are the first type. As you might guess from the name, they run directly in the

> terminal. Their advantage is that they are very lightweight and run even on tiny

Reviewer: ```suggestion
terminal. Their advantage is that they are very lightweight and run even on less powerful
```

processors. Also, they don’t use GUI elements, which can be beneficial if you

> want to edit files on a server (often, you cannot have a GUI there). As a

Reviewer: ```suggestion
want to edit files remotely on a machine without a graphical interface (like a server). As a
```

> disadvantage, they are not that convenient to use as a beginner.

Reviewer: "One downside of terminal editors is that they are ..."

> Besides terminal editors, there are graphical editors for a more intimate user

Reviewer: ```suggestion
Besides terminal editors, there are _graphical_ editors for a more intimate user
```

experience, which are often called **IDE** (Integrated Development Environment).
IDEs offer everything you need for development, such as **build automation** and
a graphical **debugger**. We will demonstrate this in upcoming posts. Since
editors became very powerful over the past years, the distinction between an
editor and an IDE is often unclear.

> Some IDEs are specific to one application. For example, in the image above, you

Reviewer: Huh, I don't see the image..?

> Some IDEs are specific to one application. For example, in the image above, you

Reviewer: ```suggestion
Some IDEs are specific to one programming language. For example, in the image above, you
```

can see PyCharm, which was developed for Python. Over time, IDEs that target
multiple (or nearly all) purposes, such as Atom or VS code, became very popular.

> Usually, they come as very basic IDE and extensions add functionality for

Reviewer: ```suggestion
Usually, they come as a very basic IDE, and _extensions_ allow the user to tailor the editor to suit their needs.
```

different needs. We will install extensions for VS code later in this post.

Usually, all kinds of editors offer a high degree of customization. You can
change colors, add shortcuts, and much more. We provide you with our

> customization, but most likely, you will find your own customizations over time.

Reviewer: ```suggestion
customization, but most likely, you will find your own customizations over time.
```
"We will provide you with our own customizations as a starting point..."

> What else can you expect from a good IDE? **Indexing** is very useful for larger

Reviewer: This should be a new paragraph. Also move the previous sentences to the paragraph about extensions above.

Author: recheck

code bases. Imagine your code has more than a thousand lines divided over

> multiple files. You want to see the definition of a used function. One way would

Reviewer: ```suggestion
multiple files (which is not unrealistic!). You want to see the definition of a used function. One way would
```

be to search all files for the function, but if the function is used multiple

> times, it is not convenient to find the function definition. Because the IDE

Reviewer: ```suggestion
times across different files, searching is not a very convenient way to find the function definition. This is where indexing comes in. Because the IDE
```

indexed all code, it knows where the function definition is and can directly
open it for you. So, indexing can be described as generating an address book of

> all variables, functions, types, and so on. **Linting** checks your code while

Reviewer: ```suggestion
all variables, functions, types, and so on. Another cool feature is **Linting**. Linting checks your code while
```

you type. It can detect errors like a typo in a variable right away. Don't worry
if you don't have a picture of these features yet. It will become clear once you
start coding.

> Let's continue with the installation.

Reviewer: ```suggestion
Now, let's get started and install VS Code!
```

we want to use something else called _Snap_. Feel free to read this
<a href="https://en.wikipedia.org/wiki/Snap_(software)"> Wikipedia</a> article
to learn more about it. Why don't we use Pacman? Simply because Pacman doesn't
provide VS code. If you want to get installation instructions, it’s often a good
idea to visit the program’s homepage.

First, we use Pacman to install Snap:

    sudo pacman -S snapd

Snap is a system service that runs in the background. We have to enable it:

    sudo systemctl enable snapd
    sudo systemctl start snapd

Finally, we need to create a link:

    sudo ln -s /var/lib/snapd/snap /snap

Sounds all like black magic to you? Well, often it does for us as well :-D Let's
keep things simple here: If you want to install something, you will find
instructions with the help of Google.

Before we can install VS code, we have to reboot the computer. This ensures that
everything is set up correctly for Snap to work. Install VS code:

    sudo snap install code --classic

That's it!

## Starting VS code for the first time

You can run VS code either from the application menu in Manjaro or from the
terminal:

    code &

> Once VS code is started, you see a _Welcome_ screen, from where you directly can

Reviewer: ```suggestion
Once VS code is started, you see a _Welcome_ screen, from where you can directly
```

create files and start coding. But let's deal with some settings first. On the
left bar, you can see several icons. If you hover over them with your mouse, you
get some explanation.

Let's start with the fifth icon from the top, which is labeled **Extensions**.
Once you click on the icon, you can look for extensions on the marketplace. Good
extensions can improve your coding experience a lot. So, it is wise to spend
some time to find the extensions that meet your needs. But since it is hard to
know your needs as a beginner, we prepared you a list of our favorite
extensions. Before we show you how to install them, click on the gear wheel icon
at the bottom and select **Settings**. Here, you can customize almost
everything: Colors, cursor styles, auto-saving, and much more. When you have
time, it’s worth checking out what is possible. But since we want to get coding

> fast, we share our settings with you.

Reviewer: ```suggestion
fast, we will share our settings with you.
```

VS code supports so-called **Profiles**. Profiles store extensions and settings.
It's possible to have multiple profiles to target different projects. For

> example, you might have one profile for Python and one for web programming using

Reviewer: ```suggestion
example, you might have one profile for Python and another one for web programming using
```
Also, don't think the "using HTML..." is really needed. Simply "web programming" should be enough.

HTML and CSS. It is also possible to export/import profiles, so it's super easy
to share settings with friends and colleagues. From our
[_Getting Started_](https://github.com/bitsandvolts/getting_started/blob/master/config_files/vs_code/bitsandvolts.code-profile)
repository, you can download our default profile. Click on the gear wheel icon
and select _Profiles -> Import Profile_, as shown in the image below. Select the
file you downloaded from our repo and import everything. It will take a moment
to download all extensions. Note that we might update our profile from time to
time.

<figure class="img-center" style="width:80%">
  <img src="/assets/images/gs_00009_vs_code.png" title="VS Code" alt="VS Code">
</figure>

Once everything is installed, you can find new icons added by extensions. If you
are interested, you can again click on the extensions icon. Now, you can find a
list of installed extensions. You can click on each extension to read what they

> are for. But it’s also fine to continue with our next post, where we start

Reviewer: The "But it's also fine..." sentence could be removed, as you also mention in the bottom that the next post is about Python.

programming in Python. We will explain further features of VS code when we use

> them in programming.

Reviewer: ```suggestion
them in our upcoming programming posts.
```

## Summary

We hope you were able to install VS code :-) Let's summarize a few things:

- There are terminal editors and IDEs
- Some IDEs are specific to one programming language
- Some IDEs can be used for multiple languages
- Extensions add further functionality to VS code
- VS code supports profiles to switch between different settings

> In our next post, we can start with our first programming language: Python :-)

Reviewer: ```suggestion
In our next post, we will start with our first programming language: Python :-)
```
